pkg/controller/nginxingress/nginx_ingress_controller.go

```diff
@@ -148,6 +148,13 @@ func (n *nginxIngressControllerReconciler) Reconcile(ctx context.Context, req ct
 	controllerDeployment = resources.Deployment
 	ingressClass = resources.IngressClass
 
+	if &nginxIngressController.Spec.DefaultSSLCertificate != nil {
```

Comment:
You're turning this into a pointer when it's not already. This will always not be nil.

We need the changes to the crd I said above then this will already be a pointer.